twitch4j-ktx/build.gradle.kts

```diff
@@ -0,0 +1,29 @@
+// In this section you declare the dependencies for your production and test code
+dependencies {
+	// Twitch4J Modules
+	// We use compileOnly so using this library does not require all modules.
+	// This does mean that using code that references modules not available will crash.
+	// This shouldn't be an issue as most, if not all, code are extension functions.
+	compileOnly(project(":twitch4j"))
+
+	// Kotlin coroutines
+	api(group = "org.jetbrains.kotlinx", name = "kotlinx-coroutines-core")
+	testImplementation(group = "org.jetbrains.kotlinx", name="kotlinx-coroutines-test")
+	testImplementation(project(":twitch4j"))
+}
+
+tasks.javadoc {
+	options {
+		title = "Twitch4J (v${version}) - Kotlin extension functions"
+		windowTitle = "Twitch4J (v${version}) - Kotlin extension functions"
+	}
+}
```

Comment:
Didn't even catch it but it is bunch of externals. We can exclude some links if it is no needed.
```suggestion
tasks {
  dokkaJavadoc {
    dokkaSourceSets {
      configureEach {
        // Adding bunch of external links for outputs
        externalDocumentationLink("https://javadoc.io/doc/org.jetbrains/annotations/latest/")
        externalDocumentationLink("https://javadoc.io/doc/com.github.ben-manes.caffeine/caffeine/2.9.2")
        externalDocumentationLink("https://commons.apache.org/proper/commons-configuration/javadocs/v1.10/apidocs/")
        externalDocumentationLink("https://javadoc.io/doc/com.github.vladimir-bukhtoyarov/bucket4j-core/7.0.0/")
        externalDocumentationLink("https://square.github.io/okhttp/4.x/okhttp/")
        externalDocumentationLink("https://javadoc.io/doc/com.github.philippheuer.events4j/events4j-core/latest/")
        externalDocumentationLink("https://javadoc.io/doc/com.github.philippheuer.events4j/events4j-handler-simple/latest/")
        externalDocumentationLink("https://javadoc.io/doc/com.github.philippheuer.credentialmanager/credentialmanager/latest/")
        externalDocumentationLink("https://javadoc.io/doc/io.github.openfeign/feign-slf4j/latest/")
        externalDocumentationLink("https://javadoc.io/doc/io.github.openfeign/feign-okhttp/latest/")
        externalDocumentationLink("https://javadoc.io/doc/io.github.openfeign/feign-jackson/latest/")
        externalDocumentationLink("https://javadoc.io/doc/io.github.openfeign/feign-hystrix/latest/")
        externalDocumentationLink("https://javadoc.io/doc/io.github.openfeign/feign-hystrix/latest/")
        externalDocumentationLink("https://netflix.github.io/Hystrix/javadoc/")
        externalDocumentationLink("https://takahikokawasaki.github.io/nv-websocket-client/")
        externalDocumentationLink("https://commons.apache.org/proper/commons-io/apidocs/")
        externalDocumentationLink("https://commons.apache.org/proper/commons-lang/apidocs/")
        externalDocumentationLink("https://www.javadoc.io/doc/org.slf4j/slf4j-api/1.7.33")
        externalDocumentationLink("https://fasterxml.github.io/jackson-databind/javadoc/2.13/")
        externalDocumentationLink("https://fasterxml.github.io/jackson-core/javadoc/2.13/")
        externalDocumentationLink("https://fasterxml.github.io/jackson-annotations/javadoc/2.13/")
        externalDocumentationLink("https://fasterxml.github.io/jackson-modules-java8/javadoc/datetime/2.13/")
        externalDocumentationLink("https://projectlombok.org/api/")
        externalDocumentationLink("https://twitch4j.github.io/javadoc/")
        externalDocumentationLink("https://kotlin.github.io/kotlinx.coroutines/") // including coroutines
      }
    }
  }
  javadocJar {
    // appending javadoc task using dokkaJavadoc
    dependsOn(dokkaJavadoc)
    from(dokkaJavadoc.get().outputDirectory.get())
  }
  // all `KotlinCompile` tasks must have `jvmTarget` set to `1.8` because we import `kotlin-stdlib-jdk8`
  withType<org.jetbrains.kotlin.gradle.tasks.KotlinCompile> {
    kotlinOptions.jvmTarget = "1.8"
  }
}
```

Reply:
I did not forget about this btw! I just want to better dive into and research this before I apply it :)

Reply:
I plan to move common buildscript into a reusable gradle plugin sometime, for javadoc i made auto-linking of all dependencies using javadoc.io if possible -> https://github.com/PhilippHeuer/gradle-projectconfiguration/blob/main/plugin/src/main/kotlin/me/philippheuer/projectcfg/modules/documentation/JavadocDocumentation.kt#L71-L105

I'm sure the same method could be used with dokka. So the links would be temporary until the plugin can link this automatically. https://github.com/twitch4j/twitch4j/pull/498

It's also good to keep this out of the gradle files because dependabot keeps updating the javadoc links and not the actual dependencies as for example in this pr: https://github.com/twitch4j/twitch4j/pull/522